MapboxNavigation/NavigationViewController.swift

```diff
@@ -529,7 +530,7 @@ open class NavigationViewController: UIViewController {
                 let directions = routeController.directions
                 let route = routeController.routeProgress.route
                 let navigationViewController = NavigationViewController(for: route, directions: directions, locationManager: locationManager)
-                
+                navigationViewController.shouldManageApplicationIdleTimer = false
```

Comment:
1. The user connects the phone to CarPlay. The application on the phone hasn’t begun navigation yet, so `UIApplication.isIdleTimerDisabled` was been false. This code sets `NavigationViewController.shouldManageApplicationIdleTimer` to false, and `CarPlayManager.application(_:didConnectCarInterfaceController:to:)` sets `UIApplication.isIdleTimerDisabled` to true.
1. The user disconnects the phone from CarPlay. `CarPlayManager.application(_:didDisconnectCarInterfaceController:from:)` sets `UIApplication.isIdleTimerDisabled` to false. NavigationViewController is still visible on the phone.
1. After a little while, the phone goes dark because `UIApplication.isIdleTimerDisabled` is false.

Reply:
NavigationViewController is still visible on the phone.

Reply:
@1ec5 yes, but we don't want the power-saving mode turned off unless it's required for the mapping experience. After disconnection from CarPlay, we certainly need to set thee idle state disabled set to `true`.

Reply:
https://github.com/mapbox/mapbox-navigation-ios/pull/1694#discussion_r217901723

Reply:
> we don't want the power-saving mode turned off unless it's required for the mapping experience